x-pack/solutions/security/plugins/security_solution/server/lib/entity_analytics/privilege_monitoring/privilege_monitoring_data_client.ts

```diff
@@ -285,4 +312,254 @@ export class PrivilegeMonitoringDataClient {
 
     return this.opts.auditLogger?.log(event);
   }
+
+  /** Privileged User Sync Orchestration
+   *  These methods coordinate syncing users from monitoring sources.
+   */
+
+  public async plainIndexSync() {
+    // get all monitoring index source saved objects of type 'index'
+    const indexSources: MonitoringEntitySourceDescriptor[] =
+      await this.monitoringIndexSourceClient.findByIndex();
+    if (indexSources.length === 0) {
+      this.log('debug', 'No monitoring index sources found. Skipping sync.');
+      return;
+    }
+    const allStaleUsers: PrivMonBulkUser[] = [];
+
+    for (const source of indexSources) {
+      // eslint-disable-next-line no-continue
+      if (!source.indexPattern) continue; // if no index pattern, skip this source
+      const index: string = source.indexPattern;
+
+      try {
+        const batchUserNames = await this.getAllUsernamesFromIndex({
+          indexName: index,
+          kuery: source.filter?.kuery,
+        });
+        // collect stale users
+        const staleUsers = await this.findStaleUsersForIndex(index, batchUserNames);
+        allStaleUsers.push(...staleUsers);
+      } catch (error) {
+        if (
+          error?.meta?.body?.error?.type === 'index_not_found_exception' ||
+          error?.message?.includes('index_not_found_exception')
+        ) {
+          this.log('warn', `Index "${index}" not found — skipping.`);
+          // eslint-disable-next-line no-continue
+          continue;
+        }
+        this.log('error', `Unexpected error during sync for index "${index}": ${error.message}`);
+      }
+    }
+    // Soft delete stale users
+    this.log('debug', `Found ${allStaleUsers.length} stale users across all index sources.`);
+    if (allStaleUsers.length > 0) {
+      const ops = this.bulkOperationsForSoftDeleteUsers(allStaleUsers, this.getIndex());
+      await this.esClient.bulk({ body: ops });
+    }
+  }
+
+  public async getAllUsernamesFromIndex({
```

Comment:
`getAllUsernamesFromIndex` does much more than its name suggests. A ' get `function should only retrieve data. This one adds data to an index.